pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -845,7 +845,17 @@ func requestConfigByURL(getconfigCtx *getconfigContext, url string,
 
 	cfgRetval = inhaleDeviceConfig(getconfigCtx, config, fromController)
 	if cfgRetval != configOK {
-		log.Errorf("inhaleDeviceConfig failed: %d", cfgRetval)
+		log.Warnf("inhaleDeviceConfig failed or skipped: %s", cfgRetval.String())
```

Comment:
can't we just save it unconditionally just after (or before?) inhaleDeviceConfig? Neither of the functions we call after this 'if' will be affected and we have a single place where we save a protobuf message. it makes logic cleaner IMO

Reply:
besides if the semantic of configOk assumes you do not save "broken" config we need return values from inhaleDeviceConfig that are not ambiguous